aten/src/ATen/ZeroTensorFallback.cpp

```diff
@@ -93,6 +93,8 @@ namespace at {
     m.impl("add.Scalar", torch::CppFunction::makeFallthrough());
     m.impl("copy_", torch::CppFunction::makeFallthrough());
     m.impl("clone", torch::CppFunction::makeFallthrough());
+    m.impl("dot", torch::CppFunction::makeFallthrough());
```

Comment:
Can we avoid adding these in the fallthrough way given that we plan to move away from that in the future?

Not blocking this PR! just asking for next ones.

Reply:
actually this should be fine because `_efficientzerotensor` is only implemented for `CPU` and `CUDA` backends so if any other backend tries to call for example `torch.dot(tensor, dual_obj)`, it'll error out because they don't have an implementation for `at::_efficientzerotensor`

Reply:
ok yeah so that sounds good. Alternately now that @bdhirsh 's dispatch key PR is merged, we can register these fallthroughs only for `ZeroTensorCPU` and `ZeroTensorCUDA` and call fallback otherwise.

Reply:
maybe next week :')